### PR TITLE
Added resume form

### LINKED ISSUE #34: Update resume
Create form

### PR DESCRIPTION
**Pull request by TomD011099**
#34 
View page will be done on another pr

### CODE REVIEW

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/payload/dtos/ExperienceCreateDto.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import java.time.LocalDate;
+
+public class ExperienceCreateDto {
+
+    @NotBlank
+    private String jobTitle;
+
+    private String functionCategory;
+
+    // TODO check with new Company entity?
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/payload/dtos/ExperienceCreateDto.java]: Experience doesn't have to be with a company in the system.

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/security/SecurityService.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                    .findByEmail(currentUserName)
+                    .orElseThrow(() -> new UsernameNotFoundException("User Not Found with email: " + currentUserName));
+        }
+        return null;
+    }
+
+    public JobSeeker getJobSeeker() {
+        Account account = getCurrentUser();
+        if (account instanceof JobSeeker) {
+            return jobSeekerRepository.getById(account.getId());
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/security/SecurityService.java]: Why would you get something you already have from the database?

```Java
if (account instanceof JobSeeker seeker) return seeker;
```

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/security/SecurityService.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        } else {
+            throw new UsernameNotFoundException("The logged in user isn't a jobseeker");
+        }
+    }
+
+    public Recruiter getRecruiter() {
+        Account account = getCurrentUser();
+        if (account instanceof Recruiter) {
+            return recruiterRepository.getById(account.getId());
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/security/SecurityService.java]: Why would you get something you already have from the database?

```Java
if (account instanceof Recruiter recruiter) return recruiter;
```

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/repositories/JobSeekerRepository.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package world.inetum.realdolmen.realjobs.repositories;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import world.inetum.realdolmen.realjobs.entities.JobSeeker;
+import world.inetum.realdolmen.realjobs.entities.Resume;
+
+public interface JobSeekerRepository extends JpaRepository<JobSeeker, Long> {
+
+}
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/repositories/JobSeekerRepository.java]: Should not be needed, `AccountRepository` already covers it.

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/repositories/RecruiterRepository.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package world.inetum.realdolmen.realjobs.repositories;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import world.inetum.realdolmen.realjobs.entities.JobSeeker;
+import world.inetum.realdolmen.realjobs.entities.Recruiter;
+
+public interface RecruiterRepository extends JpaRepository<Recruiter, Long> {
+
+}
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/repositories/RecruiterRepository.java]: Should not be needed, `AccountRepository` already covers it.

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+    @Autowired
+    public ResumeController(ResumeService resumeService, ResumeMapper resumeMapper, AccountMapper accountMapper) {
+        this.resumeService = resumeService;
+        this.resumeMapper = resumeMapper;
+        this.accountMapper = accountMapper;
+    }
+
+    @PostMapping("/skill")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    public List<SkillReadDto> addSkill(@Valid @RequestBody SkillCreateDto newSkill) {
+        return resumeService
+                .addSkill(resumeMapper.toEntity(newSkill))
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @DeleteMapping("/skill/{id}")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+    public List<SkillReadDto> removeSkill(@Valid @PathVariable("id") long id) {
+        return resumeService
+                .removeSkill(id)
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @GetMapping("/skill")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    public List<SkillReadDto> getSkills() {
+        return resumeService
+                .getSkills()
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @PostMapping("/language")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+    public List<LanguageReadDto> addLanguage(@Valid @RequestBody LanguageCreateDto newLanguage) {
+        return resumeService
+                .addLanguage(resumeMapper.toEntity(newLanguage))
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @DeleteMapping("/language/{id}")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+    public List<LanguageReadDto> removeLanguage(@Valid @PathVariable("id") long id) {
+        return resumeService
+                .removeLanguage(id)
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @GetMapping("/language")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+    public List<LanguageReadDto> getLanguages() {
+        return resumeService
+                .getLanguages()
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @PostMapping("/education")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+    public List<EducationReadDto> addEducation(@Valid @RequestBody EducationCreateDto newEducation) {
+        return resumeService
+                .addEducation(resumeMapper.toEntity(newEducation))
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @DeleteMapping("/education/{id}")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+    public List<EducationReadDto> removeEducation(@Valid @PathVariable("id") long id) {
+        return resumeService
+                .removeEducation(id)
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @GetMapping("/education")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+    public List<EducationReadDto> getEducationList() {
+        return resumeService
+                .getEducationList()
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @PostMapping("/experience")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 16]
```diff
@@ -0,0 +1,182 @@
...
+    public List<ExperienceReadDto> addExperience(@Valid @RequestBody ExperienceCreateDto newExperience) {
+        return resumeService
+                .addExperience(resumeMapper.toEntity(newExperience))
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @DeleteMapping("/experience/{id}")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+    public List<ExperienceReadDto> removeExperience(@Valid @PathVariable("id") long id) {
+        return resumeService
+                .removeExperience(id)
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @GetMapping("/experience")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+    public List<ExperienceReadDto> getExperienceList() {
+        return resumeService
+                .getExperienceList()
+                .stream()
+                .map(resumeMapper::toDto)
+                .toList();
+    }
+
+    @PostMapping("/summary")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 19]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    @GetMapping("/summary")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
+    public SingleValueDto<String> getSummary() {
+        return new SingleValueDto<>(resumeService.getSummary());
+    }
+
+    @PostMapping("/status")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+    public SingleValueDto<ResumeStatus> setStatus(@RequestBody @NotNull SingleValueDto<ResumeStatus> status) {
+        return new SingleValueDto<>(resumeService.setStatus(status.getValue()));
+    }
+
+    @GetMapping("/status")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java [hunk 21]
```diff
@@ -0,0 +1,182 @@
...
+    public SingleValueDto<ResumeStatus> getStatus() {
+        return new SingleValueDto<>(resumeService.getStatus());
+    }
+
+    @GetMapping("/account")
+    @PreAuthorize("hasRole('ROLE_JOBSEEKER')")
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/controllers/ResumeController.java]: imo, `@RolesAllowed("JOBSEEKER")` would be clearer

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/services/ResumeService.java [hunk 22]
```diff
@@ -0,0 +1,166 @@
...
+        return resume.getStatus();
+    }
+
+    public ResumeStatus getStatus() {
+        JobSeeker currentUserJs = securityService.getJobSeeker();
+        logger.info("User {} fetched their status", currentUserJs.getId());
+        return currentUserJs.getResume().getStatus();
+    }
+
+    public Account getAccount() {
```
Comment (DeKleineKobini) [inline inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/services/ResumeService.java:163]: Why not return a `JobSeeker`?

File: inetum-realjobs-frontend/src/app/app.module.ts [hunk 23]
```diff
@@ -11,10 +11,45 @@ import {UniversalAppInterceptor} from "src/app/interceptor/universalAppIntercept
...
+import {ChipModule} from "primeng/chip";
+import {LanguageListComponent} from './components/resume/details/language-list/language-list.component';
+import {StatusFormComponent} from './components/resume/forms/status-form/status-form.component';
+import {SummaryFormComponent} from './components/resume/forms/summary-form/summary-form.component';
+import {InputTextareaModule} from "primeng/inputtextarea";
+import {ConfirmPopupModule} from "primeng/confirmpopup";
+import {ConfirmationService} from 'primeng/api';
+import {ConfirmDialogModule} from "primeng/confirmdialog";
+import {PasswordModule} from "primeng/password";
+import {PanelModule} from "primeng/panel";
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/app.module.ts]: PrimeNG imports should be done in `primeng-imports.ts`

File: inetum-realjobs-frontend/src/app/app.module.ts [hunk 24]
```diff
@@ -11,10 +11,45 @@ import {UniversalAppInterceptor} from "src/app/interceptor/universalAppIntercept
...
+import {EducationFormComponent} from './components/resume/forms/education-form/education-form.component';
+import {ExperienceDetailsComponent} from './components/resume/details/experience-details/experience-details.component';
+import {CardModule} from "primeng/card";
+import {RippleModule} from "primeng/ripple";
+import {StyleClassModule} from "primeng/styleclass";
+import {ExperienceListComponent} from './components/resume/details/experience-list/experience-list.component';
+import {EducationDetailsComponent} from './components/resume/details/education-details/education-details.component';
+import {EducationListComponent} from './components/resume/details/education-list/education-list.component';
+import {SkillListComponent} from './components/resume/details/skill-list/skill-list.component';
+import {ChipModule} from "primeng/chip";
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/app.module.ts]: PrimeNG imports should be done in `primeng-imports.ts`

File: inetum-realjobs-frontend/src/app/app.module.ts [hunk 25]
```diff
@@ -11,10 +11,45 @@ import {UniversalAppInterceptor} from "src/app/interceptor/universalAppIntercept
...
+import {InputTextModule} from "primeng/inputtext";
+import {LanguageFormComponent} from "./components/resume/forms/language-form/language-form.component";
+import {SkillFormComponent} from "./components/resume/forms/skill-form/skill-form.component";
+import {ExperienceFormComponent} from "./components/resume/forms/experience-form/experience-form.component";
+import {CheckboxModule} from "primeng/checkbox";
+import {EducationFormComponent} from './components/resume/forms/education-form/education-form.component';
+import {ExperienceDetailsComponent} from './components/resume/details/experience-details/experience-details.component';
+import {CardModule} from "primeng/card";
+import {RippleModule} from "primeng/ripple";
+import {StyleClassModule} from "primeng/styleclass";
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/app.module.ts]: PrimeNG imports should be done in `primeng-imports.ts`

File: inetum-realjobs-frontend/src/app/app.module.ts [hunk 26]
```diff
@@ -11,10 +11,45 @@ import {UniversalAppInterceptor} from "src/app/interceptor/universalAppIntercept
...
 import {NotFoundComponent} from "./components/not-found/not-found.component";
 import {VacancyDetailsComponent} from "./components/vacancy-details/vacancy-details.component";
 import {ErrorHandlerComponent} from "./components/error-handler/error-handler.component";
 import {PrimengImportsModule} from "./primeng-imports";
+import {ResumeCreateComponent} from "./components/resume/resume-create/resume-create.component";
+import {InputTextModule} from "primeng/inputtext";
+import {LanguageFormComponent} from "./components/resume/forms/language-form/language-form.component";
+import {SkillFormComponent} from "./components/resume/forms/skill-form/skill-form.component";
+import {ExperienceFormComponent} from "./components/resume/forms/experience-form/experience-form.component";
+import {CheckboxModule} from "primeng/checkbox";
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/app.module.ts]: PrimeNG imports should be done in `primeng-imports.ts`

File: inetum-realjobs-frontend/src/app/app.module.ts [hunk 27]
```diff
@@ -11,10 +11,45 @@ import {UniversalAppInterceptor} from "src/app/interceptor/universalAppIntercept
...
+import {MessageModule} from "primeng/message";
+import {ToastModule} from "primeng/toast";
+import {InputMaskModule} from "primeng/inputmask";
+import {UrlInterceptor} from "./interceptor/url.interceptor";
 import {NotFoundComponent} from "./components/not-found/not-found.component";
 import {VacancyDetailsComponent} from "./components/vacancy-details/vacancy-details.component";
 import {ErrorHandlerComponent} from "./components/error-handler/error-handler.component";
 import {PrimengImportsModule} from "./primeng-imports";
+import {ResumeCreateComponent} from "./components/resume/resume-create/resume-create.component";
+import {InputTextModule} from "primeng/inputtext";
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/app.module.ts]: PrimeNG imports should be done in `primeng-imports.ts`

File: inetum-realjobs-frontend/src/app/app.module.ts [hunk 28]
```diff
@@ -11,10 +11,45 @@ import {UniversalAppInterceptor} from "src/app/interceptor/universalAppIntercept
...
 import {RegisterComponent} from "./components/register/register.component";
 import {BrowserAnimationsModule} from "@angular/platform-browser/animations";
+import {DropdownModule} from "primeng/dropdown";
+import {CalendarModule} from "primeng/calendar";
+import {FileUploadModule} from "primeng/fileupload";
+import {RadioButtonModule} from "primeng/radiobutton";
+import {MessagesModule} from "primeng/messages";
+import {MessageModule} from "primeng/message";
+import {ToastModule} from "primeng/toast";
+import {InputMaskModule} from "primeng/inputmask";
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/app.module.ts]: PrimeNG imports should be done in `primeng-imports.ts`

File: inetum-realjobs-frontend/src/app/app.routing.ts [hunk 29]
```diff
@@ -14,6 +15,7 @@ const appRoutes: Routes = [
   {path: "register", component: RegisterComponent},
   {path: "vacancy/search", component: FilterFormComponent},
   {path: "vacancy/:id", component: VacancyDetailsComponent},
+  {path: "resume/create", component: ResumeCreateComponent, canActivate: [AuthGuardService]},
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/app.routing.ts]: Shouldn't it be only available to job seekers? This guard only requires authentication iirc

File: inetum-realjobs-frontend/src/app/components/resume/details/education-details/education-details.component.ts [hunk 30]
```diff
@@ -0,0 +1,25 @@
...
+})
+export class EducationDetailsComponent implements OnInit {
+
+  @Input() education: Education;
+
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/education-details/education-details.component.ts]: Empty constructor should be removed.

File: inetum-realjobs-frontend/src/app/components/resume/details/education-details/education-details.component.ts [hunk 31]
```diff
@@ -0,0 +1,25 @@
...
+
+  ngOnInit(): void {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/education-details/education-details.component.ts]: Empty `ngOnInit` should be removed. Don't forget to remove `implements OnInit` for the class.

File: inetum-realjobs-frontend/src/app/components/resume/details/education-list/education-list.component.ts [hunk 32]
```diff
@@ -0,0 +1,25 @@
...
+  @Input() educationList: Education[];
+
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/education-list/education-list.component.ts]: Empty `ngOnInit` should be removed. Don't forget to remove `implements OnInit` for the class.

File: inetum-realjobs-frontend/src/app/components/resume/details/experience-details/experience-details.component.ts [hunk 33]
```diff
@@ -0,0 +1,32 @@
...
+  @Input() experience: Experience;
+
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/experience-details/experience-details.component.ts]: Empty `ngOnInit` should be removed. Don't forget to remove `implements OnInit` for the class.

File: inetum-realjobs-frontend/src/app/components/resume/details/experience-list/experience-list.component.ts [hunk 34]
```diff
@@ -0,0 +1,25 @@
...
+  @Input() experienceList: Experience[];
+
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/experience-list/experience-list.component.ts]: Empty `ngOnInit` should be removed. Don't forget to remove `implements OnInit` for the class.

File: inetum-realjobs-frontend/src/app/components/resume/details/experience-list/experience-list.component.html [hunk 35]
```diff
@@ -0,0 +1,9 @@
+<div class="experience-list-comp">
+  <div *ngFor="let exp of experienceList; let i = index">
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/experience-list/experience-list.component.html]: `i` seems unused

File: inetum-realjobs-frontend/src/app/components/resume/details/education-list/education-list.component.html [hunk 36]
```diff
@@ -0,0 +1,9 @@
+<div class="education-list-comp">
+  <div *ngFor="let edu of educationList; let i = index">
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/education-list/education-list.component.html]: `i` seems unused

File: inetum-realjobs-frontend/src/app/components/resume/details/education-list/education-list.component.ts [hunk 37]
```diff
@@ -0,0 +1,25 @@
...
+})
+export class EducationListComponent implements OnInit {
+
+  @Input() educationList: Education[];
+
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/education-list/education-list.component.ts]: Empty constructor should be removed.

File: inetum-realjobs-frontend/src/app/components/resume/details/experience-details/experience-details.component.ts [hunk 38]
```diff
@@ -0,0 +1,32 @@
...
+})
+export class ExperienceDetailsComponent implements OnInit {
+
+  @Input() experience: Experience;
+
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/experience-details/experience-details.component.ts]: Empty constructor should be removed.

File: inetum-realjobs-frontend/src/app/components/resume/details/experience-list/experience-list.component.ts [hunk 39]
```diff
@@ -0,0 +1,25 @@
...
+})
+export class ExperienceListComponent implements OnInit {
+
+  @Input() experienceList: Experience[];
+
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/experience-list/experience-list.component.ts]: Empty constructor should be removed.

File: inetum-realjobs-frontend/src/app/components/resume/details/language-list/language-list.component.ts [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+  styleUrls: ['./language-list.component.scss']
+})
+export class LanguageListComponent implements OnInit {
+
+  @Input() languages: Language[];
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/language-list/language-list.component.ts]: Empty constructor should be removed.

File: inetum-realjobs-frontend/src/app/components/resume/details/language-list/language-list.component.ts [hunk 41]
```diff
@@ -0,0 +1,40 @@
...
+
+  ngOnInit(): void {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/language-list/language-list.component.ts]: Empty `ngOnInit` should be removed. Don't forget to remove `implements OnInit` for the class.

File: inetum-realjobs-frontend/src/app/components/resume/details/skill-list/skill-list.component.ts [hunk 42]
```diff
@@ -0,0 +1,40 @@
...
+
+  @Input() skills: Skill[];
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/skill-list/skill-list.component.ts]: Empty `ngOnInit` should be removed. Don't forget to remove `implements OnInit` for the class.

File: inetum-realjobs-frontend/src/app/components/resume/details/skill-list/skill-list.component.ts [hunk 43]
```diff
@@ -0,0 +1,40 @@
...
+  styleUrls: ['./skill-list.component.scss']
+})
+export class SkillListComponent implements OnInit {
+
+  @Input() skills: Skill[];
+  @Input() deleteButton: boolean;
+  @Output() deleteEvent = new EventEmitter<number>();
+
+  constructor() {
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/details/skill-list/skill-list.component.ts]: Empty constructor should be removed.

File: inetum-realjobs-frontend/src/app/components/resume/resume-create/resume-create.component.ts [hunk 44]
```diff
@@ -0,0 +1,55 @@
...
+      skills: [],
+    }
+  }
+
+  onSuccess() {
+  }
+
+  onError(error: string) {
+    this.messageService.add({key: "tl", severity: "error", summary: "Error", detail: error});
+  }
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/components/resume/resume-create/resume-create.component.ts]: Everything in there looks to be unused in here?

File: inetum-realjobs-frontend/src/app/guard/jobseeker.guard.ts [hunk 45]
```diff
@@ -0,0 +1,20 @@
...
+})
+export class JobseekerGuard implements CanActivate {
+
+  constructor(private auth: AuthenticationService, private router: Router) {
+  }
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    if (this.auth.)
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/guard/jobseeker.guard.ts]: ?

File: inetum-realjobs-frontend/src/app/interceptor/url.interceptor.ts [hunk 46]
```diff
@@ -0,0 +1,19 @@
...
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class UrlInterceptor implements HttpInterceptor {
+
+  constructor() {}
```
Comment (DeKleineKobini) [inline inetum-realjobs-frontend/src/app/interceptor/url.interceptor.ts:13]: Empty constructor should be removed.

**Review by DeKleineKobini — DISMISSED** [review]

**Comment by DeKleineKobini** [comment]
It works, but I'm getting errors in the console upon load.

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/security/SecurityService.java (hunk 2 above)
Comment (TomD011099) [reply inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/security/SecurityService.java]: The problem is that I need the resume and when fetching it from the accountreposiory, that doens't exist there, so I have to refetch

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/security/SecurityService.java (hunk 3 above)
Comment (TomD011099) [reply inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/security/SecurityService.java]: The problem is that if you need the vacancy and when you fetch from the accountreposiory, that doens't exist there, so I have to refetch

File: inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/services/ResumeService.java (hunk 22 above)
Comment (TomD011099) [reply inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/services/ResumeService.java:163]: All required data was available in Account, didn't think it was necessary.

File: inetum-realjobs-frontend/src/app/components/resume/forms/experience-form/experience-form.component.ts [hunk 47]
```diff
@@ -0,0 +1,187 @@
...
+  //
+  //   if (currentJob) {
+  //     return false;
+  //   }
+  //   if (!startDate || !endDate) {
+  //     return false;
+  //   }
+  //
+  //   return this.isDateBefore(endDate, startDate);
+  // }
```
Comment (redasaada) [inline inetum-realjobs-frontend/src/app/components/resume/forms/experience-form/experience-form.component.ts]: Should we keep commented code on dev ?

Comment (TomD011099) [reply inetum-realjobs-frontend/src/app/components/resume/forms/experience-form/experience-form.component.ts]: Fixed

**Review by redasaada — APPROVED** [review]

**Review by MouadOutmane — APPROVED** [review]

**Pull request merged**